MidasFrontend/ArduinoHelper.cxx

```diff
@@ -0,0 +1,195 @@
+#include "ArduinoHelper.h"
+#include "TestStandMessages.h"
+#include "Gantry.h"
+
+#include "LinuxSerialDevice.h"
+#include "TestStandCommHost.h"
+
+#include "shared_defs.h"
+
+#include "midas.h"
+
+#include <stdio.h>
+#include <math.h>
+
+#define BAUD_RATE           115200
+#define MSG_RECEIVE_TIMEOUT 5000
+
+LinuxSerialDevice device;
+TestStandCommHost comm(device);
+
+// Gantry Mechanical Properties
+const float pulley_diameter_mm   = 17;
+const float dist_per_rev_mm      = 2 * M_PI * (pulley_diameter_mm / 2);
+const float mm_cts_ratio         = dist_per_rev_mm / ENCODER_COUNTS_PER_REV;
+const float mm_steps_ratio       = dist_per_rev_mm / MOTOR_STEPS_PER_REV;
+
+// Operational Bounds
+const float gantry_x_min_mm      = 0.0;
+const float gantry_x_max_mm      = 1200.0; // max rail is 1219 mm
+const float gantry_y_min_mm      = 0.0;
+const float gantry_y_max_mm      = 1200.0;
+const float gantry_vel_min_mm_s  = 0.0; 
+const float gantry_vel_max_mm_s  = 10.0;
+
+/*****************************************************************************/
+/*                             PRIVATE FUNCTIONS                             */
+/*****************************************************************************/
+
+static bool validate_move_params(float *dest_mm, float *vel_mm_s)
+{
+    if (dest_mm[AXIS_X] < gantry_x_min_mm || dest_mm[AXIS_X] > gantry_x_max_mm) {
+        cm_msg(
+            MERROR,
+            "validate_move_params",
+            "Destination on x-axis should be between %f mm and %f mm inclusive.",
+            gantry_x_min_mm, gantry_x_max_mm);
+        return false;
+    }
+
+    if (dest_mm[AXIS_Y] < gantry_y_min_mm || dest_mm[AXIS_Y] > gantry_y_max_mm) {
+        cm_msg(
+            MERROR,
+            "validate_move_params",
+            "Destination on y-axis should be between %f mm and %f mm inclusive.",
+            gantry_y_min_mm, gantry_y_max_mm);
+        return false;
+    }
+
+    if (vel_mm_s[AXIS_X] < gantry_vel_min_mm_s || vel_mm_s[AXIS_X] > gantry_vel_max_mm_s
```

Comment:
Instead of erroring out when a user specifies velocity > max, it might be more user friendly to give a WARNING, but execute command with hardware max velocity.

Reply:
Same for min velocity

Reply:
I think we're better off erroring out. I don't think we should ever execute any movements under parameters that we think the user intended. What if they typed an extra zero so the velocity is way out of bounds and now we're running at max speed when they only wanted to run slowly. I think there's lots of scenarios where this could go wrong.